src/dune_file.ml

```diff
@@ -230,20 +154,23 @@ module Pkg = struct
 end
 
 module Pp : sig
-  type t = private string
+  type t = private Lib_name.t
   val of_string : string -> t
   val to_string : t -> string
   val compare : t -> t -> Ordering.t
+  val to_lib_name : t -> Lib_name.t
 end = struct
-  type t = string
+  type t = Lib_name.t
+
+  let to_lib_name s = s
 
   let of_string s =
     assert (not (String.is_prefix s ~prefix:"-"));
-    s
+    Lib_name.of_string_exn s
```

Comment:
This doesn't look right indeed, we should call of_string and raise a located error in case of error. We can also add `~loc` argument to `of_string_exn`.